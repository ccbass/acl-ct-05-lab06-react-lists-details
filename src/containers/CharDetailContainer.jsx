import React, { useState } from 'react'
import {useParams} from "react-router-dom";
import { useAvatarFetchOneChar } from '../components/characters/CharHooks';
// import {useAvatarFetchChars} from '../components/characters/CharHooks'
// import CharList from '../components/characters/CharList'



function CharDetailContainer() {
    const {id} = useParams()
    const {loading, data} = useAvatarFetchOneChar(id)

    if (data){
        return (
            <div>
                <h2>Details for: {data.name}</h2>
                {data.name}
    
            </div>
        )
    }
    else{
        return null
    }
}



export default CharDetailContainer
