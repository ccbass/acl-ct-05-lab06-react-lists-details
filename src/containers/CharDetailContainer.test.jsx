import React from 'react';
import { render, screen } from '@testing-library/react';
import CharDetailContainer from './CharDetailContainer';
import { MemoryRouter, Route } from 'react-router-dom';

import { avatarDetailData } from '../../fixtures/avatar-data';
import { rest } from 'msw';
import { setupServer } from 'msw/node';


const server = setupServer(
  rest.get(
    'https://last-airbender-api.herokuapp.com/api/v1/characters/5cf5679a915ecad153ab68d5',
    (req, res, ctx) => {
      return res(
        ctx.json(avatarDetailData)
      );
    }
  )
);

describe('CharDetailContainer container tests', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('detailed character tests', async () => {
    render(
      <MemoryRouter initialEntries={['/5cf5679a915ecad153ab68d5']}>
        <Route path="/:id"> <CharDetailContainer /> </Route>
      </MemoryRouter>
    );

    // Check for loading text to appear before API resolves
    screen.getByText('L O A D I N G');
  
    // Check that 1 image card loads and displays in DOM
    const image = await screen.findAllByRole('img');
    expect(image).toHaveLength(1);
    expect(image).toMatchSnapshot();

    // Check if character data is in DOM
    const character = await screen.findByText('Earth Kingdom Air Force');
    expect(character).toBeInTheDocument();
    const characterJob = await screen.findByText('Airman');
    expect(characterJob).toBeInTheDocument();

  });
      
      
});
