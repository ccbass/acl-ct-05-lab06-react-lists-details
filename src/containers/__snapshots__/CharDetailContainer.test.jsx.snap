// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharDetailContainer container tests detailed character tests 1`] = `
Array [
  <img
    src="https://vignette.wikia.nocookie.net/avatar/images/6/6f/Arik.png/revision/latest?cb=20140808184002"
  />,
]
`;
